slash_commands/mine-sweeper.js

```diff
@@ -0,0 +1,115 @@
+const { ApplicationCommandOptionType } = require('discord.js');
+const { serverIcons } = require('../config.js');
+
+const iconMap = {
+  '-2' : serverIcons.mineBombShiny ?? '💥',
+  '-1' : serverIcons.mineBomb ?? '🧨',
+  '0' : '🟩',
+  '1' : '1️⃣',
+  '2' : '2️⃣',
+  '3' : '3️⃣',
+  '4' : '4️⃣',
+  '5' : '5️⃣',
+  '6' : '6️⃣',
+  '7' : '7️⃣',
+  '8' : '8️⃣',
+};
+const shinyChance = 54;
+const getBombType = () => !Math.floor(Math.random() * shinyChance) ? -2 : -1;
+const revealTime = 3 * 60; // Three minutes
+
+module.exports = {
+  name        : 'minesweeper',
+  aliases     : ['mine', 'mine-sweeper'],
+  description : 'Generate a game of Minesweeper to play',
+  args        : [
+    {
+      name: 'width',
+      type: ApplicationCommandOptionType.Integer,
+      description: 'How big you want the board to be',
+      required: false,
+    },
+    {
+      name: 'bombs',
+      type: ApplicationCommandOptionType.Integer,
+      description: 'How many bombs on the field',
+      required: false,
+    },
+    {
+      name: 'height',
+      type: ApplicationCommandOptionType.Integer,
+      description: 'How big you want the board to be',
+      required: false,
+    },
```

Comment:
Might be worth switching bombs and height, so player can just specify heigh + width rather than needing to specify bombs